libvast/src/system/index.cpp

```diff
@@ -172,13 +175,15 @@ index_state::index_state(index_actor::pointer self)
 caf::error index_state::load_from_disk() {
   // We dont use the filesystem actor here because this function is only
   // called once during startup, when no other actors exist yet.
-  if (!exists(dir)) {
+  std::error_code err{};
+  const auto file_exists = std::filesystem::exists(dir, err);
+  if (!file_exists) {
```

Comment:
Still need to check `err` here.

Reply:
I'm not sure we do based on my reading at [CppRef](https://en.cppreference.com/w/cpp/filesystem/exists). I have checked `err` in previous uses of this function in this codebase, but after some more reading, I don't think it's required. My understanding is the function  will return `false` still but also populate `err` which could be used in log messages for example if desired.

Reply:
I verified that you are indeed correct on this one.

Reply:
> I verified that you are indeed correct on this one.

Thanks for double checking. I'll put up a PR that doesn't check the extra error in the one or two other call sites I've added it in form previous PRs.